pkg/registry/core/service/rest.go

```diff
@@ -586,6 +534,54 @@ func (rs *REST) initClusterIP(service *api.Service) (bool, error) {
 	return false, nil
 }
 
+func (rs *REST) initNodePort(service *api.Service, nodePortOp *portallocator.PortAllocationOperation) error {
+	assignNodePorts := shouldAssignNodePorts(service)
```

Comment:
I write a test case and find that the existing logic is incorrect, unfortunately. Thought we have `shouldAssignNodePorts`, a `should-not` type, e.g. `ClusterIP` still can be allocated a specified NodePort in `Create`. Fortunately, a `should-not` type of service with specified NodePort will fail the validation ahead of `Create`. So it's safe here. But we should fix this here anyway, IMO.